apps/studio/components/interfaces/HomePageActions.tsx

```diff
@@ -17,45 +19,51 @@ import {
 import { Input } from 'ui-patterns/DataInputs/Input'
 
 interface HomePageActionsProps {
-  search: string
-  filterStatus: string[]
   hideNewProject?: boolean
-  viewMode?: 'grid' | 'table'
   showViewToggle?: boolean
-  setSearch: (value: string) => void
-  setFilterStatus: (value: string[]) => void
-  setViewMode?: (value: 'grid' | 'table') => void
 }
 
 export const HomePageActions = ({
-  search,
-  filterStatus,
   hideNewProject = false,
-  viewMode,
   showViewToggle = false,
-  setSearch,
-  setFilterStatus,
-  setViewMode,
 }: HomePageActionsProps) => {
   const { slug } = useParams()
   const projectCreationEnabled = useIsFeatureEnabled('projects:create')
 
+  const [search, setSearch] = useQueryState('search', parseAsString.withDefault(''))
+  const [filterStatus, setFilterStatus] = useQueryState(
+    'status',
+    parseAsArrayOf(parseAsString, ',').withDefault([])
+  )
+  const [viewMode, setViewMode] = useLocalStorageQuery(LOCAL_STORAGE_KEYS.PROJECTS_VIEW, 'grid')
+
   return (
     <div className="flex items-center justify-between">
       <div className="flex items-center gap-2">
         <Input
```

Comment:
The search over the network can be a little slow, which sometimes makes it look like nothing is happening. Worth putting a loading state on this?

Reply:
i'll add a loader here as an indicator for the card view 🙂 
<img width="586" height="94" alt="image" src="https://github.com/user-attachments/assets/e1e8e0cf-af8e-4d1f-af17-6a21b6e1c24f" />